aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,658 @@
+- Has an HTML Helper alternative ``Html.BeginForm`` and ``Html.BeginRouteForm``
+
+Sample:
+
+.. literalinclude::   forms/sample/final/Views/Demo/RegisterFormOnly.cshtml
+  :language: HTML
+
+The Form Tag Helper above generates the following HTML:
+ 
+.. code-block:: HTML
+
+  <form method="post" action="/Demo/Register">
+    <!-- Input and Submit elements -->
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+   </form>
+  
+The MVC runtime generates the ``action`` attribute value from the Form Tag Helper attributes ``asp-controller`` and ``asp-action``. The Form Tag Helper also generates a hidden `Request Verification Token <http://www.asp.net/mvc/overview/security/xsrfcsrf-prevention-in-aspnet-mvc-and-web-pages>`__ to prevent cross-site request forgery (when used with the ``[ValidateAntiForgeryToken]`` attribute in the HTTP Post action method). Protecting a pure HTML Form from cross-site request forgery is very difficult, the Form Tag Helper provides this service for you.
+
+Many of the views in the  *Views/Account* folder (generated when you create a new web app with *Individual User Accounts*) contain the `asp-route-returnurl <http://docs.asp.net/en/latest/mvc/views/working-with-forms.html#the-form-tag-helper>`__ attribute: 
+
+.. code-block:: HTML
+  :emphasize-lines: 2
+  
+  <form asp-controller="Account" asp-action="Login" 
+    asp-route-returnurl="@ViewData["ReturnUrl"]" 
+    method="post" class="form-horizontal" role="form">
+
+:Note: With the built in templates, ``returnUrl`` is only populated automatically when you try to access an authorized resource but are not authenticated or authorized. When you attempt an unauthorized access, the security middleware redirects you to the login page with the ``returnUrl`` set.
+
+Using a named route
+^^^^^^^^^^^^^^^^^^^
+
+The ``asp-route`` Tag Helper attribute can also generate markup for the HTML ``action`` attribute. An app with a :doc:`route </fundamentals/routing>`  named ``register`` could use the following markup for the registration page:
+ 
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterRoute.cshtml 
+  :language: HTML
+  :emphasize-lines: 4
+
+The Input Tag Helper
+---------------------
+ 
+The Input Tag Helper binds an HTML `<input> <https://www.w3.org/wiki/HTML/Elements/input>`__ element to a model expression in your razor view.
+
+Syntax:
+
+.. code-block:: HTML
+
+  <input asp-for="<Expression Name>" /> 
+
+The Input Tag Helper:
+
+- Generates the ``id`` and ``name`` HTML attributes for the expression name specified in the ``asp-for`` attribute.  ``asp-for="Property1.Property2"`` is equivalent to ``m => m.Property1.Property2``, that is the attribute value literally is part of an expression. The name of the expression is what's used for the ``asp-for`` attribute value.
```

Comment:
The first paragraph of the **Expression Names** section is
```
The ``asp-for`` attribute value is a `ModelExpression <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/ModelExpression/index.html>`__
and the right hand side of a lambda expression. Therefore, ``asp-for="Property1"`` becomes ``m => m.Property1``
in the generated code which is why you don't need to prefix with ``Model``. You can use the "@" character to start
an inline expression and move before the ``m.``:
```

That looks like duplication to me. Further your last sentence is contradictory since you just said the `asp-for` value is literally the RHS of a lambda expression.